packages/expo-updates/plugin/src/withUpdatesIOS.ts

```diff
@@ -0,0 +1,215 @@
+): ExpoPlist {
+  const newExpoPlist = {
+    ...expoPlist,
+    [Config.ENABLED]: getUpdatesEnabled(config),
+    [Config.CHECK_ON_LAUNCH]: getUpdatesCheckOnLaunch(config),
+    [Config.LAUNCH_WAIT_MS]: getUpdatesTimeout(config),
+  };
+
+  const updateUrl = getUpdateUrl(config, username);
+  if (updateUrl) {
+    newExpoPlist[Config.UPDATE_URL] = updateUrl;
+  } else {
+    delete newExpoPlist[Config.UPDATE_URL];
+  }
+
+  return setVersionsConfig(config, newExpoPlist);
+}
+
+export function setVersionsConfig(config: ExpoConfigUpdates, expoPlist: ExpoPlist): ExpoPlist {
+  const newExpoPlist = { ...expoPlist };
+
+  const runtimeVersion = getRuntimeVersion(config);
+  const sdkVersion = getSDKVersion(config);
+  if (runtimeVersion) {
+    delete newExpoPlist[Config.SDK_VERSION];
+    newExpoPlist[Config.RUNTIME_VERSION] = runtimeVersion;
+  } else if (sdkVersion) {
+    delete newExpoPlist[Config.RUNTIME_VERSION];
+    newExpoPlist[Config.SDK_VERSION] = sdkVersion;
+  } else {
+    delete newExpoPlist[Config.SDK_VERSION];
+    delete newExpoPlist[Config.RUNTIME_VERSION];
+  }
+
+  return newExpoPlist;
+}
+
+function formatConfigurationScriptPath(projectRoot: string): string {
+  const buildScriptPath = resolveFrom.silent(projectRoot, CREATE_MANIFEST_IOS_PATH);
+
+  if (!buildScriptPath) {
+    throw new Error(
+      "Could not find the build script for iOS. This could happen in case of outdated 'node_modules'. Run 'npm install' to make sure that it's up-to-date."
+    );
+  }
+
+  return path.relative(path.join(projectRoot, 'ios'), buildScriptPath);
+}
+
+interface ShellScriptBuildPhase {
+  isa: 'PBXShellScriptBuildPhase';
+  name: string;
+  shellScript: string;
+  [key: string]: any;
+}
+
+export function getBundleReactNativePhase(project: XcodeProject): ShellScriptBuildPhase {
+  const shellScriptBuildPhase = project.hash.project.objects.PBXShellScriptBuildPhase as Record<
+    string,
+    ShellScriptBuildPhase
+  >;
+  const bundleReactNative = Object.values(shellScriptBuildPhase).find(
+    buildPhase => buildPhase.name === '"Bundle React Native code and images"'
+  );
+
+  if (!bundleReactNative) {
+    throw new Error(`Couldn't find a build phase "Bundle React Native code and images"`);
+  }
+
+  return bundleReactNative;
+}
+
+export function ensureBundleReactNativePhaseContainsConfigurationScript(
+  projectRoot: string,
+  project: XcodeProject
+): XcodeProject {
+  const bundleReactNative = getBundleReactNativePhase(project);
+  const buildPhaseShellScriptPath = formatConfigurationScriptPath(projectRoot);
+
+  if (!isShellScriptBuildPhaseConfigured(projectRoot, project)) {
+    // check if there's already another path to create-manifest-ios.sh
+    // this might be the case for monorepos
+    if (bundleReactNative.shellScript.includes(CREATE_MANIFEST_IOS_PATH)) {
+      bundleReactNative.shellScript = bundleReactNative.shellScript.replace(
+        new RegExp(`(\\\\n)(\\.\\.)+/node_modules/${CREATE_MANIFEST_IOS_PATH}`),
```

Comment:
```suggestion
        new RegExp(`(\\\\n)(\\.\\./)+node_modules/${CREATE_MANIFEST_IOS_PATH}`),
```
not 100% sure this is correct, but the way it is now seems like it would match `....../node_modules` but not `../../../node_modules`

Reply:
copied it verbatim from the `expo/config-plugins` package.

Reply:
well, feel free to ignore this suggestion if you want. I don't know where the code in expo/config-plugins came from (it wasn't me that wrote it)

Reply:
https://github.com/expo/expo-cli/pull/3103